agent/api/container/container.go

```diff
@@ -910,3 +945,56 @@ func (c *Container) GetStopTimeout() time.Duration {
 
 	return time.Duration(c.StopTimeout) * time.Second
 }
+
+func (c *Container) SetRestartAttempts(count RestartCount) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.RestartAttempts = count
+}
+
+func (c *Container) GetRestartAttempts() RestartCount {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	return c.RestartAttempts
+}
+
+func (c *Container) IncrementRestartAttempts() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.RestartAttempts += 1
+}
+
+func (c *Container) CanMakeRestartAttempt() bool {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	return c.RestartPolicy == Always ||
+		c.RestartPolicy == OnFailure &&
+		c.RestartAttempts < c.RestartMaxAttempts
+}
+
+func (c *Container) IsAutoRestartNonEssentialContainer() bool {
+	return !c.IsEssential() && c.RestartPolicy != NEVER
+}
+
+func (c *Container) IsDesiredToRestartWhenReceivingStopped() bool {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	defer func() {
+		c.DesiredToRestartWhenReceivingStopped = false
+	}()
+	return c.DesiredToRestartWhenReceivingStopped
+}
+
+func (c *Container) SetDesiredToRestartWhenReceivingStopped() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.DesiredToRestartWhenReceivingStopped = true
+}
+
+func (c *Container) SetDefaultRestartMaxAttemptsOnFailure() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if c.RestartPolicy == OnFailure && c.RestartMaxAttempts == 0 {
```

Comment:
Why are we setting this to `RestartMaxAttempts` to `DefaultRestartMaxAttemptsOnFailure` when it 0?

Reply:
By default, if user doesn't specify `RestartMaxAttempts` for policy: `OnFailure', we'll keep restarting the container if its exit code is not 0.